systest/backup/filesystem/backup_test.go

```diff
@@ -40,19 +40,88 @@ import (
 )
 
 var (
-	copyBackupDir = "./data/backups_copy"
-	restoreDir    = "./data/restore"
-	testDirs      = []string{restoreDir}
-
-	alphaBackupDir = "/data/backups"
-
+	copyBackupDir   = "./data/backups_copy"
+	restoreDir      = "./data/restore"
+	testDirs        = []string{restoreDir, exporterDir}
+	exporterDir     = "data/exporter"
+	alphaBackupDir  = "/data/backups"
+	oldBackupDir    = "/data/to_restore"
 	alphaContainers = []string{
 		"alpha1",
 		"alpha2",
 		"alpha3",
 	}
 )
 
+func sendRestoreRequest(t *testing.T, location string) int {
+	if location == "" {
+		location = "/data/backup"
+	}
+	params := testutil.GraphQLParams{
+		Query: `mutation restore($location: String!) {
+			restore(input: {location: $location}) {
+				code
+				message
+				restoreId
+			}
+		}`,
+		Variables: map[string]interface{}{
+			"location": location,
+		},
+	}
+	resp := testutil.MakeGQLRequest(t, &params)
+	resp.RequireNoGraphQLErrors(t)
+
+	var restoreResp struct {
+		Restore struct {
+			Code      string
+			Message   string
+			RestoreId int
+		}
+	}
+
+	require.NoError(t, json.Unmarshal(resp.Data, &restoreResp))
+	require.Equal(t, restoreResp.Restore.Code, "Success")
+	require.Greater(t, restoreResp.Restore.RestoreId, 0)
+	return restoreResp.Restore.RestoreId
+}
+
+// This test takes a backup and then restores an old backup in a cluster incrementally.
+// Next, cleans up the cluster and tries restoring the backups above.
+func TestBackupOfOldRestore(t *testing.T) {
+	dirSetup(t)
+	copyOldBackupDir(t)
+	conn, err := grpc.Dial(testutil.SockAddr, grpc.WithInsecure())
+	require.NoError(t, err)
+	dg := dgo.NewDgraphClient(api.NewDgraphClient(conn))
+
+	ctx := context.Background()
+
+	require.NoError(t, dg.Alter(ctx, &api.Operation{DropAll: true}))
+	time.Sleep(2 * time.Second)
```

Comment:
Please confirm that these sleeps are good enough by running the tests multiple times. Otherwise these can be flaky.

Reply:
Good point. Ran it 10 times locally.